p2p/host/routed/routed.go

```diff
@@ -119,6 +128,15 @@ func (rh *RoutedHost) RemoveStreamHandler(pid protocol.ID) {
 }
 
 func (rh *RoutedHost) NewStream(ctx context.Context, p peer.ID, pids ...protocol.ID) (inet.Stream, error) {
+	// check if we need to find some addresses for the peer through the routing system
+	if len(rh.Network().ConnsToPeer(p)) == 0 && len(rh.Peerstore().Addrs(p)) == 0 {
+		addrs, err := rh.findPeerAddrs(ctx, p)
+		if err != nil {
+			return nil, err
+		}
+		rh.Peerstore().AddAddrs(p, addrs, pstore.TempAddrTTL)
+	}
+
```

Comment:
Updated to use Connect; so yeah, both shorter and easier to understand with the right comment.